src/negotiation-status-manager.ts

```diff
@@ -0,0 +1,152 @@
+import EventEmitter from 'eventemitter3';
+
+type Events = {
+  updated: [];
+};
+
+export enum NegotiationState {
+  NOT_RESOLVED = 0,
+  RESOLVED
+};
+
+export enum ResolveState {
+  UNKNOWN = 0,
+  RESOLVED,
+  FAILED
+};
+
+
+export class NegotiationStatus {
+  sequence : number[] = [];
+  defunct : boolean = false;
+  rejected : boolean = false;
+  forfeited : boolean = false;
+};
+
+export class NegotiationStatusData {
+  has_terminal : boolean = false;
+  base : NegotiationStatus = new NegotiationStatus();
+  terminal : NegotiationStatus = new NegotiationStatus();
+}
+
+export class NegotiationConflict {
+  public participant_ids_to_names : Record<string, string> = {};
+  public participant_ids_to_status : Record<string, NegotiationStatusData> = {};
+  public resolved : ResolveState = ResolveState.UNKNOWN;
+};
+
+export default class NegotiationStatusManager extends EventEmitter<Events> {
+  constructor(private _backend_ws ?: WebSocket) {
+    super();
+  }
+
+  updateNegotiationStatus(msg : any) {
```

Comment:
Let's try not to use `any`, `any` it's our last resource. You can create custom types to accept different kind of inputs. 
type Test = string|number|record<string, string>

Reply:
these functions are only used in one place and I'd rather not make another type, so inlined functions c3fc9b7

Reply:
Cool! Just a comment (nit) on  https://github.com/osrf/romi-dashboard/commit/c3fc9b79067a3838fad71fb0cbb2b9642755f9d3

```
 const conflict_version : number = msg["conflict_version"];
 const conflict_version_str = conflict_version.toString()
```
conflict_version -> conflictVersion

Reply:
nice catch, align code style for file 1912262